### PR TITLE
Added `amenities` to `restStop`

### PR DESCRIPTION
**Pull request by abhishek1508**
Fixes: https://github.com/mapbox/mapbox-java/issues/1445

### CODE REVIEW

**Comment by Guardiola31337** [comment]
@abhishek1508 why is this a `Draft`? Is it ready for review?

**Comment by Guardiola31337** [comment]
Let's wait to merge here until we have public docs.

**Comment by LukasPaczos** [comment]
Could you add `DirectionsResponse`/`DirectionsRoute` tests as well to verify serialization/deserialization within response objects?

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Amenities.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ * Only available on the {@link DirectionsCriteria#PROFILE_DRIVING} profile.
+ */
+@AutoValue
+public abstract class Amenities extends DirectionsJsonObject {
+
+  /**
+   * The type of amenities such as gas, restaurants, shopping, bank, atm etc.
+   * Note that adding new possible types is not considered a breaking change.
+   */
+  public abstract String type();
```
Comment (Guardiola31337) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Amenities.java:22]: We should add `@NonNull` / `@Nullable` annotation.

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Amenities.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+     * @param type amenity type
+     */
+    public abstract Builder type(String type);
+
+    /**
+     * The name of the amenity. Optionally included if data is available.
+     *
+     * @param name amenity name
+     */
+    public abstract Builder name(@Nullable String name);
```
Comment (Guardiola31337) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Amenities.java:96]:
```suggestion
     */
    @NonNull
    public abstract Builder name(@Nullable String name);
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Amenities.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * The brand name of the amenity. Optionally included if data is available.
+     *
+     * @param brand amenity brand name
+     */
+    public abstract Builder brand(@Nullable String brand);
```
Comment (Guardiola31337) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Amenities.java:103]:
```suggestion
     */
    @NonNull 
    public abstract Builder brand(@Nullable String brand);
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Amenities.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * Build a new {@link Amenities} object.
+     *
+     * @return a new {@link Amenities} using the provided values in this builder
+     */
+    public abstract Amenities build();
```
Comment (Guardiola31337) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Amenities.java:110]:
```suggestion
     */
    @NonNull 
    public abstract Amenities build();
```

File: services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RestStop.java [hunk 5]
```diff
@@ -93,6 +101,13 @@ public abstract static class Builder extends DirectionsJsonObject.Builder<Builde
      */
     public abstract Builder name(@Nullable String name);
 
+    /**
+     * The list of amenities at the rest stop if available.
+     *
+     * @param amenities list of amenities
+     */
+    public abstract Builder amenities(@Nullable List<Amenities> amenities);
```
Comment (Guardiola31337) [inline services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RestStop.java:109]:
```suggestion
     */
    @NonNull 
    public abstract Builder amenities(@Nullable List<Amenities> amenities);
```

We can also add the above missing annotations 👍

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/AmenitiesTest.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  private Amenities getDefault() {
+    return Amenities.builder()
+        .type("restaurant")
+        .name("starbucks")
+        .build();
+  }
+
+  @Test
+  public void testSerializableObject() throws Exception {
```
Comment (Guardiola31337) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/AmenitiesTest.java:24]: We should also remove `test` prefix from tests names, it's implicit with the usage of `@Test` annotation.

Same thing applies to other tests below 👇

File: services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/AmenitiesTest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    Amenities amenities = Amenities.builder()
+        .type("restaurant")
+        .name("starbucks")
+        .build();
+    byte[] serialized = TestUtils.serialize(amenities);
+    assertEquals(amenities, deserialize(serialized, Amenities.class));
```
Comment (Guardiola31337) [inline services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/AmenitiesTest.java:30]: Could we structure the test following the Triple A?

Same thing applies to other tests below 👇

Comment (abhishek1508) [reply services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/AmenitiesTest.java]: Okay I forgot to remove the term `test` from a couple of tests. I have done that now. @Guardiola31337

**Comment by LukasPaczos** [comment]
We should not have merged the PR since the documentation (and I presume data) is not yet available. We will not include it with this week's Nav SDK releases and we need to track it for the next stable one, and revert if needed.

**Comment by abhishek1508** [comment]
Yeah I merged it by mistake.

**Pull request merged**